prediction_market_agent/agents/microchain_agent/deploy.py

```diff
@@ -23,6 +26,7 @@ def run(self, market_type: MarketType) -> None:
         agent: Agent = build_agent(
             market_type=market_type,
             model=self.model,
+            system_prompt=TRADING_AGENT_SYSTEM_PROMPT,  # Use pre-learned system prompt until the prompt-fetching from DB is implemented.
```

Comment:
After the agent updates its system prompt, it needs to be saved in the DB and then automatically fetches. This will bring some complication so I'd leave that for the next PR.

Right now, DeployableAgent is using `trader` prompt right away (instead of always learning from scratch) and the learning prompt is used only in the streamlit demo to showcase it.

Reply:
Sorry, also don't follow here - my understanding was this
-> General agent performs a bunch of actions, incl. learning but hopefully trading. It stores its actions in the DB, as well as the open positions are also queryable (e.g. via `getUserPositions`).
-> The prompt was continuously updated during the run (appending history), now the agent has the freedom to update it (this PR). OK.
-> In my head, once the run is over, the prompt is lost - i.e. a new run must start with the original prompt. However (this is the important part), by calling `getPastLearnings`, the agent can see what he learned previously (from the activities in the current table) and hence accelerate its learning.

This is all to say - I don't think we should store the prompts in the DB. What do you think?

Reply:
We need to. The system prompt (how the agent should behave) and the history of user prompts (actions that the agent took) are two different concepts.

Another thing I wanted to propose today is renaming `RememberPastLearnings` to something like `RememberPastActions`, because even in the description of that class there is `Use this function to fetch information about the actions`.

Reply:
As an example, let's say the agent starts with the system prompt `Learn, survive and evolve` and it evolved to a system prompt:

```
Act as a trader agent in prediction markets to maximise your profit.

Research markets, buy tokens you consider undervalued, and sell tokens that you
hold and consider overvalued.
```

(this is a copy-paste of the original system prompt, but good for this example, it could evolve to something similar)

We don't want him next day to start again as `Learn, evolve survive`, it already learned trading and stuff, so we want the agent to continue, not always start from 0.